backend/tests/test_critical_regressions.py

```
@@ -0,0 +1,217 @@
+        time.sleep(0.05)
+        with state_lock:
+            active -= 1
+        return {"prediction": "real", "confidence": 90, "framesAnalyzed": 1}
+
+    monkeypatch.setattr(app_module.inference, "analyze_file", fake_analyze)
+
+    def worker():
+        try:
+            barrier.wait(timeout=1.0)
+            app_module._run_inference("x.jpg", "image", 1.0)
+        except BaseException as exc:  # surface worker failures in the test thread
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `BaseException`

(BLE001)